drv/vsc7448/src/lib.rs

```diff
@@ -691,8 +691,23 @@ impl<'a, R: Vsc7448Rw> Vsc7448<'a, R> {
         Ok(())
     }
 
-    /// Implements the VLAN scheme described in RFD 250.
-    pub fn configure_vlan_strict(&self) -> Result<(), VscError> {
+    /// Configures VLANs for a production-style system.
+    ///
+    /// For details, see RFD 250, but in brief:
+    ///
+    /// - VLANs in the form 0x1YY are configured for each downstream port.
+    ///   Ports active on a given VLAN are selected by the given function `f`;
+    ///   typically, each VLAN will contain a downstream port and the uplink
+    ///   port.
+    /// - Downstream ports send and receive untagged frames, and apply their
+    ///   VLAN tag on packet ingress.
+    /// - The uplink port (49) sends and receives packets with one VLAN tag, and
+    ///   uses that packet to select which VLAN (i.e. which downstream port)
+    ///   should receive that packet.  The VLAN tag is stripped on egress.
+    fn configure_vlan_with_mask<F: Fn(u8) -> u64>(
+        &self,
+        f: F,
```

Comment:
I think because the closures passed here don't actually close over anything, it doesn't need to be generic and can be

```rust
f: fn(u8) -> u64,
```

with no changes at the call sites.

Reply:
Neat! I had assumed that wouldn't work (because it was a closure), but apparently [non-capturing closures will automatically be coerced into plain `fn`'s](https://rust-lang.github.io/rfcs/1558-closure-to-fn-coercion.html).